paperoni/sources/scrapers/openalex.py

```diff
@@ -0,0 +1,385 @@
+                            else []
+                        ),
+                    ),
+                    affiliations=[
+                        Institution(
+                            name=author_inst["display_name"],
+                            category=INSTITUTION_CATEGORY_MAPPING[
+                                author_inst["type"]
+                            ],
+                            aliases=[],
+                        )
+                        for author_inst in authorship["institutions"]
+                    ],
+                )
+                for authorship in data["authorships"]
+            ],
+            releases=[
+                Release(
+                    venue=Venue(
+                        type=VENUE_TYPE_MAPPING[location["source"]["type"]],
+                        name=location["source"]["display_name"],
+                        series="",
+                        date=publication_date,
+                        date_precision=DatePrecision.day,
+                        volume=None,
+                        publisher=location["source"]["host_organization_name"],
+                        aliases=[],
+                        links=(
+                            [
+                                Link(
+                                    type="url",
+                                    link=location["landing_page_url"],
+                                )
+                            ]
+                            if location["landing_page_url"]
+                            else []
+                        )
+                        + (
+                            [Link(type="pdf", link=location["pdf_url"])]
+                            if location["pdf_url"] is not None
+                            else []
+                        ),
+                        # NB: I guess "open" mean "an Open Access version of this work is available at this location" ?
+                        open=location["is_oa"],
+                        # https://docs.openalex.org/api-entities/works/work-object/location-object#version
+                        peer_reviewed=(
+                            location["version"]
+                            in ("acceptedVersion", "publishedVersion")
+                        ),
+                    ),
+                    status=(
+                        "published"
+                        if location.get("is_published")
+                        else (
+                            "accepted"
+                            if location.get("is_accepted")
+                            else "unknown"
+                        )
+                    ),
+                    pages=None,
+                )
+                for location in release_locations
+            ],
+            topics=[
+                Topic(name=data_concept["display_name"])
+                for data_concept in data["concepts"]
+            ],
+            links=[Link(type=typ, link=ref) for typ, ref in data["ids"].items()]
```

Comment:
It seems that the ids are all URLs, which is not consistent with what Paperoni does, e.g. `data[ids][doi] == 'https://doi.org/10.48550/arxiv.2111.09266'` but we want `10.48550/arxiv.2111.09266`, the part after the `/`. I think a decent general rule would be `ref.split("/", 3)[-1]` (we keep what is after the third slash).

The same goes for an author's `id` and `orcid` when present.

Reply:
Yes, that's perfect!